src/Interpreters/Context.cpp

```diff
@@ -4288,7 +4288,7 @@ ReadSettings Context::getReadSettings() const
             "Invalid value '{}' for max_read_buffer_size", settings.max_read_buffer_size);
     }
 
-    res.local_fs_buffer_size = settings.max_read_buffer_size;
+    /* res.local_fs_buffer_size = settings.max_read_buffer_size; */
```

Comment:
Remove it altogether.

Reply:
maybe it worth to create a separate setting `max_read_buffer_size_for_local_fs`?

Reply:
Yes, let's introduce two new settings - for local and remote fs, and the old max_read_buffer_size setting will be used if they are unset.